lib/iris/coords.py

```diff
@@ -1720,11 +1730,7 @@ def points(self, points):
         # Ensure points has an ndmin of 1 and is either a numpy or lazy array.
         # This will avoid Scalar coords with points of shape () rather
         # than the desired (1,).
-        if is_lazy_data(points):
-            if points.shape == ():
-                points = points.reshape(1,)
-        else:
-            points = self._sanitise_array(points, 1)
+        points = self._sanitise_array(points, 1)
```

Comment:
I do intend to have a go at reorganising the code a bit, because I think there is a lot of common code in the points+bounds getters+setters for DimCoord+AuxCoord ( :sweat_smile: ).
I'm definitely leaving that to a separate PR -- it should be a strict refactor, and it might not work out as easily as I think -- in which case we can quite happily run with what we have.